xla/service/cpu/cpu_compiler.cc

```diff
@@ -795,6 +795,30 @@ absl::Status CpuCompiler::RunHloPassesAfterLayoutAssn(
 
   pipeline.AddPass<ReshapeDecomposer>();
 
+  // The LayoutAssignment pass may leave behind kCopy instructions which are
+  // duplicate or NOPs, so remove them with algebraic simplification and CSE.
+  // Run this to a fixed point.
+  [&pipeline = pipeline.AddPass<HloPassFix<HloPassPipeline>>(
```

Comment:
Why do we need this lambda here? Can't we just introduce a nested scope, like is done e.g. on line 783 above?

Reply:
I don't know if there was a reason to use lambda, I just moved existing code. I will change it to nested scope and see it works fine.